src/gui.rs

```diff
@@ -107,54 +93,56 @@ enum Interaction {
     SelectInstallation(Installation),
     SelectLoaderVersion(LoaderVersion),
     SelectMcVersion(MinecraftVersion),
-    EnableSnapshots(bool),
-    EnableBetas(bool),
+    ToggleSnapshots(bool),
+    ToggleBetas(bool),
     GenerateLaunchScript(bool),
     GenerateProfile(bool),
     ChangeServerLocation(PathBuf),
     BrowseServerLocation,
     DownloadServerJar(bool),
 }
 
-impl Into<Command<Message>> for Message {
-    fn into(self) -> Command<Message> {
-        async {
-            self
-        }.into()
+impl From<Message> for Command<Message> {
+    fn from(m: Message) -> Self {
+        Self::perform(async { m }, |t| t)
     }
 }
 
 #[cfg(target_os = "windows")]
 fn get_default_client_directory() -> PathBuf {
-    let mut dir = PathBuf::from(std::env::var("APPDATA").unwrap());
-    dir.push(".minecraft");
-    dir
+    PathBuf::from(std::env::var("APPDATA").unwrap()).join(".minecraft")
 }
 
 #[cfg(target_os = "macos")]
 fn get_default_client_directory() -> PathBuf {
-    let mut dir = PathBuf::from(std::env::var("HOME").unwrap());
-    dir.push("Library");
-    dir.push("Application Support");
-    dir.push("minecraft");
-    dir
+    PathBuf::from(std::env::var("HOME").unwrap())
+        .join("Library")
+        .join("Application Support")
+        .join("minecraft")
 }
 
 #[cfg(target_os = "linux")]
 fn get_default_client_directory() -> PathBuf {
-    let mut dir = PathBuf::from(std::env::var("HOME").unwrap());
-    dir.push(".minecraft");
-    dir
+    PathBuf::from(std::env::var("HOME").unwrap()).join(".minecraft")
 }
```

Comment:
Ohhh, I like this!

Reply:
Indeed! If you need to create a `mut` variable, modify it, then return it, rust usually has in-place methods to modify the variable directly and return it. (part of its functional approach ig??)